xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -107,3 +114,90 @@ async def send_reliable_submission(
+    check_fee: bool = True,
+    autofill: bool = True,
+) -> Transaction:
+    """
+    Initializes a signed transaction for a submit request.
+
+    Args:
+        transaction: the transaction or transaction blob to be signed if unsigned.
+        client: the network client to autofill from.
+        wallet: an optional wallet with which to sign the transaction. This is
+            only needed if the transaction is unsigned.
+        check_fee: an optional bolean indicating whether to check if the fee is
+            higher than the expected transaction type fee. Defaults to True.
+        autofill: an optional boolean indicating whether to autofill the
+            transaction. Defaults to True.
+
+    Returns:
+        The signed transaction.
+    """
+    if isinstance(transaction, str):
+        transaction = Transaction.from_blob(transaction)
+
+    if transaction.is_signed():
+        return transaction
+
+    if not wallet:
+        raise XRPLException(
+            "Wallet must be provided when submitting an unsigned transaction"
+        )
+
+    if check_fee:
+        await _check_fee(transaction, client)
+
+    if autofill:
+        transaction = await _autofill(transaction, client)
+
+    if transaction.signers:
+        return await sign(transaction, wallet, True, True)
+
+    return await sign(transaction, wallet, True, False)
+
+
+async def submit_and_wait(
+    transaction: Union[Transaction, str],
+    client: Client,
+    wallet: Optional[Wallet] = None,
+    *,
+    check_fee: bool = True,
+    autofill: bool = True,
+    fail_hard: bool = False,
+) -> Response:
+    """
+    Signs a transaction locally if the transaction is unsigned, then submits,
+    and verifies that it has been included in a validated ledger (or has errored/
+    will not be included for some reason).
+    `See Reliable Transaction Submission for a full explanation of this workflow
+    <https://xrpl.org/reliable-transaction-submission.html>`_
+
+    Args:
+        transaction: the signed/unsigned transaction (or transaction blob) to
+            be submitted.
+        client: the network client with which to submit the transaction.
+        wallet: an optional wallet with which to sign the transaction. This is
+            only needed if the transaction is unsigned.
+        check_fee: an optional bolean indicating whether to check if the fee is
+            higher than the expected transaction type fee. Defaults to True.
+        autofill: an optional boolean indicating whether to autofill the
+            transaction. Defaults to True.
+        fail_hard: an optional boolean. If True, and the transaction fails for
+            the initial server, do not retry or relay the transaction to other
+            servers. Defaults to False.
+
+    Returns:
+        The response from the ledger.
+    """
+    signed_transaction = await _get_signed_tx(
+        transaction, client, wallet, check_fee, autofill
+    )
+    return await send_reliable_submission(
```

Comment:
Note for later: If we make `send_reliable_submission` deprecated, we should move that logic into this function, or a helper.

Reply:
Agree, we can mention that in the deprecate ticket.